actuator/src/main/java/org/tron/core/vm/program/Program.java

```diff
@@ -1178,7 +1178,7 @@ public DataWord getBlockHash(int index) {
       BlockCapsule blockCapsule = contractState.getBlockByNum(index);
 
       if (Objects.nonNull(blockCapsule)) {
-        return new DataWord(blockCapsule.getBlockId().getBytes());
+        return new DataWord(blockCapsule.getBlockId().getBytes()).clone();
```

Comment:
The constructor of `DataWord` should be changed, instead of any caller position.

A `DataWord` must not point to any non-owned memory.

Reply:
Annyminous hackers love it